htdocs/web_portal/controllers/downtime/add_downtime.php

```diff
@@ -158,62 +158,149 @@ function draw(\User $user = null) {
         die(json_encode(array('UTC', 0)));
     }
 
-    // URL Mapping
-    // If the user wants to add a downtime to a specific site, show only that site's SEs
-    else if(isset($_GET['site'])) {
-        $site = \Factory::getSiteService()->getSite($_GET['site']);
-        if(\Factory::getRoleActionAuthorisationService()->authoriseAction(\Action::EDIT_OBJECT, $site, $user)->getGrantAction() == FALSE){
-           throw new \Exception("You don't have permission over $site");
-        }
-        $ses = $site->getServices();
-        $params = array('ses' => $ses, 'nowUtc' => $nowUtcDateTime->format('H:i T'), 'selectAll' => true);
-        show_view("downtime/add_downtime.php", $params);
-        die();
+    /**
+     * URL Mapping for `site` and `se`.
+     *
+     * If the user wants to add a downtime to a specific site and SE,
+     * show only that site SEs.
+     */
+    elseif (isset($_GET['site']) && isset($_GET['se'])) {
+        displaySiteAndSeEndpoints($user, $nowUtcDateTime);
     }
 
-    // URL Mapping
-    // If the user wants to add a downtime to a specific SE, show only that SE
-    else if(isset($_GET['se'])) {
-        $se = \Factory::getServiceService()->getService($_GET['se']);
-        $site = \Factory::getSiteService()->getSite($se->getParentSite()->getId());
-        if(\Factory::getRoleActionAuthorisationService()->authoriseAction(\Action::EDIT_OBJECT, $se->getParentSite(), $user)->getGrantAction() == FALSE){
-           throw new \Exception("You do not have permission over $se.");
-        }
+    /**
+     * URL Mapping for `site` ONLY.
+     *
+     * If the user wants to add a downtime to a specific site,
+     * show all of the site SEs.
```

Comment:
Same as above. At any given instance, there will be only one site that we will be displaying in the Select Specific site section for the above condition because the user has selected that specific site for adding downtime. It should be singular here.

Reply:
> Same as above. At any given instance, there will be only one site that we will be displaying in the Select Specific site section for the above condition because the user has selected that specific site for adding downtime. It should be singular here.

when 's is used, its to show possession, not plurality. So its just showing the SEs belong to the site, not that it will show multiple sites. or am i getting confused here. can discuss tomorrow

Reply:
The Service endpoint vs Service Endpoint endpoint terminology in GOCDB is confusing. I've left a suggestion for a comment here that I think is a little clearer: https://github.com/GOCDB/gocdb/pull/439/files#r1282082077